octodns/provider/nsone.py

```diff
@@ -0,0 +1,189 @@
+#
+#
+#
+
+from __future__ import absolute_import, division, print_function, \
+    unicode_literals
+
+from logging import getLogger
+from nsone import NSONE
+from nsone.rest.errors import ResourceException
+
+from ..record import Record
+from .base import BaseProvider
+
+
+class NsOneProvider(BaseProvider):
+    '''
+    NsOne provider
+
+    nsone:
+        class: octodns.provider.nsone.NsOneProvider
+        api_key: env/NS_ONE_API_KEY
+    '''
+    SUPPORTS_GEO = False
```

Comment:
Some have been easier than others. Haven't looked deeply enough at NS1's model & api to know what it'd involve. Don't plan to add it in this round since I won't have enough cycles to dig in. Once this is merged I'd be happy to look over a PR that adds support for it 😁.